docs/spec-files/encryption.md

```diff
@@ -44,6 +44,69 @@ For example, following the guidelines of [message types](#message-types), an enc
 
 Due to the triple Key Derivation algorithm used in ECDH-1PU, all messages sent via DIDComm have weak perfect forward secrecy without any additional security added by the transport layer. In ECDH-1PU this is achieved by encrypting the content encryption key with the output of the hash of the Ze (ECDH of ephemeral key and recipient static key) and Zs (ECDH of static sender key and recipient static key). With Ze bringing the changed derived secret in each message and Zs bringing the repudiable authenticity of each message, the resulting Z (hash of Ze and Zs) carries the properties of weak perfect forward secrecy and repudiable authenticity for each message as well.
 
+## Key IDs `kid` and `skid` headers references in the DID document
+
+Keys used by DIDComm envelopes MUST be sourced from the DIDs exchanged between two agents. Specifically, both sender and recipients keys MUST be retrieved from the DID document's `KeyAgreement` verification section as per the [DID Document Keys](https://identity.foundation/didcomm-messaging/spec/#did-document-keys) definition.
+
+When Alice is preparing an envelope intended for Bob, the packing process should use a key from both hers and Bob's DID document's `KeyAgreement` section.
```

Comment:
It will require another pr for Aries-RFCs since the pr is already [merged](https://github.com/hyperledger/aries-rfcs/pull/635).

Reply:
I'll make an editorial pass to update this later since it didn't make it into this commit. I don't believe this requires any updates to Aries-RFC as long as technically everything operates the same. This is about updating the language we use to be more standards formal where as Aries RFCs are typically developer first focused.